src/WooCommerceCheckout.php

```diff
@@ -15,22 +15,26 @@ public static function init() {
     // Removes city field from shipping calculator.
     add_filter('woocommerce_shipping_calculator_enable_city', '__return_false');
 
-    if (is_admin()) {
+    if (is_admin() || wp_doing_cron()) {
       return;
     }
 
     // Adds confirmation email field to the checkout page.
     if (
       get_option('_' . Plugin::L10N . '_checkout_email_confirmation_field') === 'yes' &&
       !is_user_logged_in() &&
-      !isset($_GET['woo-paypal-return'])
+      !isset($_GET['woo-paypal-return']) &&
+      !self::isAmazonPayV2Checkout()
     ) {
       add_filter('woocommerce_checkout_fields', __CLASS__ . '::addConfirmationEmailCheckoutField');
       add_action('woocommerce_checkout_process', __CLASS__ . '::checkConfirmationEmailField');
     }
 
     // Add checkout error messages.
     add_filter( 'woocommerce_form_field', __CLASS__ . '::woocommerceFormField', 10, 4 );
+
+    // Remove required fields while using Amazon Pay  V2.
```

Comment:
We understand the context right now, because we're working on it, but it's going to be harder in one year from now.
```suggestion
    // Remove required fields when checking out via Amazon using
    // woocommerce-gateway-amazon-payments-advanced.
```